server/src/main/java/org/candlepin/resource/JobResource.java

```diff
@@ -174,6 +181,42 @@ public SchedulerStatus setSchedulerStatus(boolean running) {
         return getSchedulerStatus();
     }
 
+    @POST
+    @Path("cron/{task}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @SuppressWarnings("unchecked")
+    public JobStatus schedule(@PathParam("task") String task) {
+
+        String className = "org.candlepin.pinsetter.tasks." + task;
+        try {
+            boolean isCron = false;
+            for (String cronJob : ConfigProperties.DEFAULT_TASK_LIST) {
+                if (className.equalsIgnoreCase(cronJob)) {
+                    isCron = true;
+                    // helps ignore case
+                    className = cronJob;
+                }
+            }
+            if (isCron) {
+                Class<? extends KingpinJob> taskClass = (Class<? extends KingpinJob>) Class
+                    .forName(className);
+                return pk.scheduleSingleJob(taskClass, Util.generateUUID());
```

Comment:
You can clean this up a bit and save yourself from line wrapping by declaring the class generically, then casting in the schedule line:

```Class taskClass = Class.forName(className);
return pk.scheduleSingleJob((Class<? extends KingpinJob>) taskClass, Util.generateUUID());```